contrib/mc-wrapper.fish.in

```diff
@@ -0,0 +1,20 @@
+set --local MC_PWD_FILE
+if set -q MC_TMPDIR
+	set MC_PWD_FILE (mktemp $MC_TMPDIR/mc.pwd.XXXXXX)
+else if set -q TMPDIR
+	set MC_PWD_FILE (mktemp $TMPDIR/mc.pwd.XXXXXX)
+else
+	set MC_PWD_FILE (mktemp /tmp/mc.pwd.XXXXXX)
+end
+
+# Propagate mc exit code outside the wrapper
+@bindir@/mc -P "$MC_PWD_FILE" $argv
```

Comment:
this will not actually propagate the exit code beyond line 13.
If we do want to propagate it (which seems like a good idea),
then we should be doing something like this (untested)
but probably for all shells

```diff
diff --git a/contrib/mc-wrapper.sh.in b/contrib/mc-wrapper.sh.in
index 399b38a892..1a8200b5de 100644
--- a/contrib/mc-wrapper.sh.in
+++ b/contrib/mc-wrapper.sh.in
@@ -6,7 +6,8 @@
 	MC_PWD_FILE="`mktemp "/tmp/mc.pwd.XXXXXX"`"
 fi
 
-@bindir@/mc -P "$MC_PWD_FILE" "$@" || true
+MC_STATUS=0
+@bindir@/mc -P "$MC_PWD_FILE" "$@" || MC_STATUS=$?
 
 if test -r "$MC_PWD_FILE"; then
 	MC_PWD="`cat "$MC_PWD_FILE"`"
@@ -18,3 +19,4 @@
 
 rm -f "$MC_PWD_FILE"
 unset MC_PWD_FILE
+eval "unset MC_STATUS; exit $MC_STATUS"
```

Reply:
I guess the comment was written assuming `set -e`, which is not the usual behavior, and even in that case we want to clean up.. though that's an edge case of an edge case